lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
```

Comment:
It might be nice to the use the `Hash.new` syntax so missing keys execute a lambda that asks for the `Protobuf::Nats` class to compute the key, that would simplify the `cached_subscription_key` method below.

We should also use both the service and method name as the key rather than just the method name.